packages/widgets/src/system-resources/component.tsx

```diff
@@ -0,0 +1,94 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { useElementSize, useListState } from "@mantine/hooks";
+
+import { clientApi } from "@homarr/api/client";
+
+import type { WidgetComponentProps } from "../definition";
+import { CombinedNetworkTrafficChart } from "./chart/combined-network-traffic";
+import { SystemResourceCPUChart } from "./chart/cpu-chart";
+import { SystemResourceMemoryChart } from "./chart/memory-chart";
+import { NetworkTrafficChart } from "./chart/network-traffic";
+import classes from "./component.module.css";
+
+const MAX_QUEUE_SIZE = 15;
+
+export default function SystemResources({ integrationIds }: WidgetComponentProps<"systemResources">) {
+  const [queue, queueHandlers] = useListState<{
+    cpu: number;
+    memory: number;
+    network: { up: number; down: number } | null;
+  }>([]);
+  const [memoryCapacityInBytes, setMemoryCapacityInBytes] = useState(0);
+
+  const { ref, width } = useElementSize();
+
+  const { data } = clientApi.widget.healthMonitoring.getSystemHealthStatus.useQuery({
+    integrationIds,
+  });
+  clientApi.widget.healthMonitoring.subscribeSystemHealthStatus.useSubscription(
+    {
+      integrationIds,
+    },
+    {
+      onData(data) {
+        const obj = {
+          cpu: data.healthInfo.cpuUtilization,
+          memory: data.healthInfo.memUsedInBytes,
+          network: data.healthInfo.network,
+        };
+        queueHandlers.setState((queue) => [...queue, obj].slice(0, MAX_QUEUE_SIZE));
+      },
+    },
+  );
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const showNetwork = queue.length === 0 ? true : queue[queue.length - 1]!.network != null;
+
+  useEffect(() => {
+    if (!data) {
+      return;
+    }
+
+    const items = data.map((itemData) => ({
+      cpu: itemData.healthInfo.cpuUtilization,
+      memory: itemData.healthInfo.memUsedInBytes,
+      network: itemData.healthInfo.network,
+    }));
+    queueHandlers.setState((queue) => [...queue, ...items].slice(0, MAX_QUEUE_SIZE));
+
+    if (data[0]) {
+      setMemoryCapacityInBytes(data[0].healthInfo.memAvailableInBytes + data[0].healthInfo.memUsedInBytes);
+    }
+  }, [data, queueHandlers]);
```

Comment:
I currently get the following error when it tries to update the charts:
<img width="972" height="245" alt="image" src="https://github.com/user-attachments/assets/9f70df52-08b7-40de-89c4-f4da258ab0dd" />

I think it would make sense to instead just define the initial value of the `useListState` after the `useQuery` and then update it only through the `useSubscription`